internal/services/storage/storage_share_resource.go

```diff
@@ -196,83 +198,91 @@ func resourceStorageShareCreate(d *pluginsdk.ResourceData, meta interface{}) err
 
 	if !features.FivePointOhBeta() {
 		storageClient := meta.(*clients.Client).Storage
-		if accountName := d.Get("storage_account_name").(string); accountName != "" {
-			shareName := d.Get("name").(string)
-			quota := d.Get("quota").(int)
-			metaDataRaw := d.Get("metadata").(map[string]interface{})
-			metaData := ExpandMetaData(metaDataRaw)
```

Comment:
Changing this means that we will use the Data Plane client even in the cases that `storage_account_id` is specified in place of `storage_account_name`, which we specifically need to avoid for this change to separate the the two APIs. If `storage_account_id` is specified, we must use exclusively the Resource Manager API.